basis/english/english.factor

```diff
@@ -141,11 +143,80 @@ PRIVATE>
         [ "s" append ]
     } cond match-case ;
 
+: singular? ( word -- ? )
+    [ singularize ] [ = ] bi ;
+
+: plural? ( word -- ? )
```

Comment:
Or maybe (even slower but less duplicative):

```factor
: plural? ( word -- ? )
    [ singularize pluralize ] [ = ] bi ;
```

Reply:
Um, I get a `Retain stack overflow` when I do that, which is pretty impressive since I don't even remember how to make one of those intentionally.

Reply:
Hmm, that should work (and it does when I paste it in).

Reply:
I restarted Factor and it runs fine, bizarre.